pgjdbc/src/main/java/org/postgresql/core/Utils.java

```diff
@@ -50,7 +50,15 @@ public static String toHexString(byte[] data) {
     // for performance measurements.
     // In OracleJDK 6u65, 7u55, and 8u40 String.getBytes(Charset) is
     // 3 times faster than other JDK approaches.
-    return str.getBytes(utf8Charset);
+    try {
+      if (str.contains("\000")) {
```

Comment:
This is reasonably cheap for short strings, but could be expensive for long strings.
Perhaps the “normal” way should be attempted first, and this modified approach attempted only in case of exception.

Are we certain all encoding goes through this method? I was under impression some encoding uses the encoder associated with the connection.

Reply:
`indexOf(0) != -1` should be faster